### ISSUE TITLE
Error/Segv when savemat called with empty-dict values

### ISSUE THREAD

**Issue by allenleetc** [frejanordsiek/hdf5storage#80]
Hi,

Apologies if this is a dup or known issue. Thanks also for this project, we are heavy users.

When calling savemat() with a dict-containing-an-empty-dict-value, I get i) an error if the target file does not exist; ii) a Segv/crash if the target file already exists. Please see below. In my testing this seems to occur with {Py2.7, Py3.5.2} x {hdf5storage v0.1.15, v0.2}.

```
Python 3.5.2 (default, Oct  8 2019, 13:06:37) 

import hdf5storage

d={'foo':{}}

hdf5storage.savemat('/path/to/foofoo.mat',d)  # first call; file does not exist yet

Traceback (most recent call last):

  File "/opt/venv/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2878, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-4-0fdda4c91617>", line 1, in <module>
    hdf5storage.savemat('/path/to/foofoo.mat',d)
  File "/opt/venv/lib/python3.5/site-packages/hdf5storage/__init__.py", line 1681, in savemat
    options=options)
  File "/opt/venv/lib/python3.5/site-packages/hdf5storage/__init__.py", line 1319, in writes
    None, options)
  File "/opt/venv/lib/python3.5/site-packages/hdf5storage/lowlevel.py", line 114, in write_data
    m.write(f, grp, name, data, type_string, options)
  File "/opt/venv/lib/python3.5/site-packages/hdf5storage/Marshallers.py", line 1614, in write
    self.write_metadata(f, grp, name, data, type_string, options)
  File "/opt/venv/lib/python3.5/site-packages/hdf5storage/Marshallers.py", line 1668, in write_metadata
    set_attribute(grp[name], 'MATLAB_fields', fs)
  File "/opt/venv/lib/python3.5/site-packages/hdf5storage/utilities.py", line 750, in set_attribute
    target.attrs.create(name, value)
  File "/opt/venv/lib/python3.5/site-packages/h5py/_hl/attrs.py", line 200, in create
    attr.write(data, mtype=htype2)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5a.pyx", line 383, in h5py.h5a.AttrID.write
  File "h5py/_proxy.pyx", line 63, in h5py._proxy.attr_rw
  File "h5py/defs.pyx", line 3405, in h5py.defs.H5Tconvert
  File "h5py/_conv.pyx", line 745, in h5py._conv.ndarray2vlen
AttributeError: '��[' object has no attribute 'dtype'


hdf5storage.savemat('/path/to/foofoo.mat',d)  # second call, file now exists

Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)
```

**Comment by frejanordsiek** [comment]
Sorry about taking so long to get back to you.

I was unable to reproduce the error on the 0.1.x or master branches with CPython 3.8.2 and h5py 2.10.0.

There is a slight chance it is related to #81 and #82 which I am in the progress of fixing (fixed for 0.1.x branch but not yet master).

**Comment by frejanordsiek** [comment]
Issues #81 and #82 are fixed now.
